authorization_feature.go

```diff
@@ -36,7 +36,31 @@ func (f *AuthorizationFeature) iAmIdentifiedAs(username string) error {
 	return nil
 }
 
+func (f *AuthorizationFeature) zebedeeRecognisesTheServiceAuthTokenAsInvalid() error {
+	f.FakeAuthService.NewHandler().Get("/serviceInstancePermissions").Reply(401).BodyString(`{ "message": "CMD permissions request denied: service account not found"}`)
+	return nil
+}
+
+func (f *AuthorizationFeature) zebedeeRecognisesTheServiceAuthTokenAsValid() error {
+	f.FakeAuthService.NewHandler().Get("/serviceInstancePermissions").Reply(200).BodyString(`{ "permissions": ["DELETE", "READ", "CREATE", "UPDATE"]}`)
+	return nil
+}
+
+func (f *AuthorizationFeature) zebedeeRecognisesTheUserTokenAsInvalid() error {
+	f.FakeAuthService.NewHandler().Get("/userInstancePermissions").Reply(401).BodyString(`{ "message": "CMD permissions request denied: session not found"}`)
+	return nil
+}
+
+func (f *AuthorizationFeature) zebedeeRecognisesTheUserTokenAsValid() error {
+	f.FakeAuthService.NewHandler().Get("/userInstancePermissions").Reply(200).BodyString(`{ "permissions": ["DELETE", "READ", "CREATE", "UPDATE"]}`)
```

Comment:
Nothing wrong with this code but these permissions are what would be returned for an `Admin` or `Publisher` user. A viewer would get empty or just `"READ"` if they were assigned to a team with preview access to a collection/dataset.

I'm not sure if that matters to your task or not but thought I would share anyway. Feel free to message me if you have questions on the permissions stuff.

Reply:
Thanks for this info. I've added it in as a comment so that people will understand that this might not be appropriate to use if their test is for a viewer.